moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -643,12 +644,21 @@ bool ompl_interface::ModelBasedPlanningContext::solve(double timeout, unsigned i
   ompl::time::point start = ompl::time::now();
   preSolve();
 
+  // Initialize a termination condition that terminates an optimizing planner if the cost function converges at
+  // a certain level. Non-optimizing planners are not affected by this since they don't produce multiple solutions.
+  // TODO(henningkayser): Implement constructor parameters
+  std::shared_ptr<CostConvergenceTerminationCondition> cctc = std::make_shared<CostConvergenceTerminationCondition>();
+  ompl_simple_setup_->getProblemDefinition()->setIntermediateSolutionCallback(
+      [cctc](const ob::Planner* planner, const std::vector<const ob::State*>& states, const ob::Cost cost) {
+        cctc->processNewSolution(states, cost);
+      });
```

Comment:
I general I approve of the change, but this should probably be a parameter to the MBPC.

Reply:
I agree, I would add a parameter but also would like to add support for different termination conditions in the long run